File: code/Java/interpreter/command/IfCommand.java
package interpreter.command;

import interpreter.expr.BoolExpr;

public class IfCommand extends Command {

    private BoolExpr cond;
    private Command thenCmds;
    private Command elseCmds;

    public IfCommand(int line, BoolExpr cond, Command thenCmds) {
        this(line, cond, thenCmds, null);
    }
    
    public IfCommand(int line, BoolExpr cond, Command thenCmds, Command elseCmds) {
        super(line);
        this.cond = cond;
        this.thenCmds = thenCmds;
        this.elseCmds = elseCmds;
    }

    public void execute() {
        if (cond.expr())
            thenCmds.execute();
        else {
            if (elseCmds != null)
                elseCmds.execute();
        }
    }

}

File: code/C++/interpreter/command/IfCommand.cpp
#include "IfCommand.h"
#include "../expr/BoolExpr.h"

IfCommand::IfCommand(int line, BoolExpr* cond, Command* thenCmds, Command* elseCmds)
	: Command(line), m_cond(cond), m_thenCmds(thenCmds), m_elseCmds(elseCmds) {
}

IfCommand::~IfCommand() {
	delete m_cond;
	delete m_thenCmds;
	if (m_elseCmds)
		delete m_elseCmds;
}

void IfCommand::execute() {
	if (m_cond->expr())
		m_thenCmds->execute();
	else {
		if (m_elseCmds)
			m_elseCmds->execute();
	}
}
